src/baldr/graphreader.cc

```diff
@@ -454,12 +459,8 @@ const GraphTile* GraphReader::GetGraphTile(const GraphId& graphid) {
         }
```

Comment:
i guess i dont remember why we even get into the `if` about 7 lines up. to me it seems we can just modify the `if` on line 452 to be `if (!tile.header() && tile_getter_) {` and then also change the `if` on line 456 to be `if (_404s.find(base) != _404s.end()) {`. this way we dont even try to get a lock when we arent configured for URL based tile fetching

Reply:
rewrote it in a way you asked